sdk/spring/azure-identity-spring-library/src/main/java/com/microsoft/azure/identity/spring/AzureIdentitySpringHelper.java

```
@@ -0,0 +1,138 @@
+        credentials.put(name, credential);
+    }
+
+    /**
+     * Get the default Azure credential.
+     *
+     * @return the default Azure credential
+     */
+    public TokenCredential getDefaultCredential() {
+        return credentials.get("");
+    }
+
+    /**
+     * Get the named credential.
+     *
+     * @param name the name.
+     * @return the named credential, or null if not found.
+     */
+    public TokenCredential getNamedCredential(String name) {
+        return credentials.get(name);
+    }
+
+    /**
+     * Populate from Environment.
+     *
+     * @param environment the environment.
+     */
+    public void populate(Environment environment) {
+        populateNamedCredential(environment, "");
+        String credentialNamesKey = AZURE_CREDENTIAL_PREFIX + "names";
+        if (environment.containsProperty(credentialNamesKey)) {
+            String[] credentialNames = environment.getProperty(credentialNamesKey).split(",");
+            for(int i=0; i<credentialNames.length; i++) {
+                populateNamedCredential(environment, credentialNames[i]);
+            }
+        }
+    }
+
+    /**
+     * Populate a named credential.
+     *
+     * @param environment the environment
+     * @param name the name.
+     */
+    private void populateNamedCredential(Environment environment, String name) {
+        
+        if (!name.equals("") && !name.endsWith(".")) {
+            name = name + ".";
+        }
+        
+        String tenantIdKey = AZURE_CREDENTIAL_PREFIX + name + "tenantId";
+        String clientIdKey = AZURE_CREDENTIAL_PREFIX + name + "clientId";
+        String clientSecretKey = AZURE_CREDENTIAL_PREFIX + name + "clientSecret";
+
+        String tenantId = environment.getProperty(tenantIdKey);
+        String clientId = environment.getProperty(clientIdKey);
+        String clientSecret = environment.getProperty(clientSecretKey);
+
+        if (tenantId != null && clientId != null && clientSecret != null) {
```

If some but not all of these are none, that would suggest an unintentional error condition. Perhaps it's worth failing loudly.

---

Done